catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -536,257 +477,69 @@ const useDirStyles = M.makeStyles((t) => ({
   },
 }))
 
-type DirProps = tagged.ValueOf<typeof FilesEntry.Dir> & {
-  prefix?: string
-  dispatch: DispatchFilesAction
-}
-
-function Dir({
-  contentIsUnknown,
-  disabled,
-  name,
-  type,
-  children,
-  prefix,
-  dispatch,
-}: DirProps) {
+type DirOwnProps = React.PropsWithChildren<{
+  name: string
+  state?: FilesEntryState
+  active?: boolean
+  empty?: boolean
+  expanded?: boolean
+  onChangeExpanded?: (expanded: boolean) => void
+  action?: React.ReactNode
+  onHeadClick?: React.MouseEventHandler<HTMLDivElement>
+}>
+
+type DirProps = DirOwnProps &
+  Omit<React.HTMLAttributes<HTMLDivElement>, keyof DirOwnProps>
+
+const Dir = React.forwardRef<HTMLDivElement, DirProps>(function Dir(
+  {
+    name,
+    state = 'unchanged',
+    active = false,
+    empty = false,
```

Comment:
As I can see exposing `empty` is redundant because it uses `children.length`

Reply:
there's a little nuance here: you may have empty children and not want to display the "<EMPTY>" placeholder  -- and that's exactly the case with the peer folders selection -- we dont support showing nested objects, so we just need to render the dir name and not its empty contents